src/simulations/results/results_plotting_utils.jl

```diff
@@ -0,0 +1,406 @@
+            Colorbar(fig[ax_row, ax_col + 1], hm)
+        
+        else    
+            Colorbar(fig[ax_row, ax_col + 1], hm)
+        end
+
+        # Set title, labels, and other attributes for current variable
+        ax.title = "$(get(title_mapping, string(var), (string(var), ""))[1]) ($(get(title_mapping, string(var), (string(var), ""))[2]))" # Fetch title from mapping, default to var if not found
+        ax.xlabel = "X coordinate"
+        ax.ylabel = "Y coordinate"
+    end
+
+    return fig
+end
+
+"""
+    plot_glacier_difference(
+        results::T, 
+        variables::Vector{Symbol}; 
+        tspan::Tuple{Float64, Float64}, 
+        metrics::Vector{String} = ["difference"]
+    ) -> Figure
+
+Plot the difference metrics for a specific glacier variable over a given time span.
+
+# Arguments
+- `results`: A custom type containing the results of a glacier simulation.
+  * Must have fields corresponding to the variables to be plotted.
+- `variables`: A vector of symbols representing the variable to be plotted. Only one variable is allowed.
+  * Possible variables: `:H` or any other data that has time evolution in it.
+- `tspan`: A tuple representing the start and end time for the simulation.
+- `metrics`: A vector of strings indicating the type of metrics to visualize. Possible values are: "hist" (histogram) and "difference". Defaults to `["difference"]`.
+
+# Returns
+- A `Figure` object containing the difference metrics for the specified glacier variable and or a histogram showing the distribution.
+
+# Notes
+- The function uses predefined titles for each variable based on a mapping any other variable just get their symbol as title.
+- Only one variable can be plotted at a time with this function.
+"""
+function plot_glacier_difference(results::T, variables::Vector{Symbol}; tspan::Tuple{Float64, Float64}, metrics::Vector{String} = ["difference"]) ::Figure where T 
+        
+        # Check if more than one variable is passed
+        if length(variables) > 1
+            error("Only one variable can be passed to this function.")
+        end
+    
+        # Check for valid metrics
+        valid_metrics = ["hist", "difference"]
+        for metric in metrics
+            if !(metric in valid_metrics)
+                error("Invalid metric: $metric. Valid metrics are: $valid_metrics")
+                return
+            end
+        end
+    
+        # Extract data for the variable
+        data = getfield(results, variables[1])
+    
+        # Check the shape of the extracted data
+        if typeof(data) ≠ Vector{Matrix{Float64}}
+            error("Only temporal quantities can be used in this function.")
+        end
+        
+        # Extract the rgi_id 
+        rgi_id = :rgi_id in fieldnames(typeof(results)) ? results.rgi_id : "none"
+        
+        # Print plot information
+        variable_title = get(title_mapping, variables[1], variables[1])
+        println("Glacier Visualization Summary")
```

Comment:
General comment, I don't think you need to print anything in the console when plotting. Plots should be self-contained and undestandable by themselves. Any text or relevant information (albeit not redundant) should go in the plot itself.

Reply:
Removed print